src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/Signing/VerifyCommand.cs

```diff
@@ -0,0 +1,82 @@
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Extensions.CommandLineUtils;
+using NuGet.Commands;
+using NuGet.Common;
+using static NuGet.Commands.VerifyArgs;
+
+namespace NuGet.CommandLine.XPlat
+{
+    internal static class VerifyCommand
+    {
+        internal static void Register(CommandLineApplication app,
+                              Func<ILogger> getLogger,
+                              Action<LogLevel> setLogLevel,
+                              Func<IVerifyCommandRunner> getCommandRunner)
+        {
+            app.Command("verify", verifyCmd =>
+            {
+                CommandArgument packagePaths = verifyCmd.Argument(
+                    "<package-paths>",
+                    Strings.VerifyCommandPackagePathDescription,
+                    multipleValues: true);
+
+                CommandOption all = verifyCmd.Option(
+                    "--all",
+                    Strings.VerifyCommandAllDescription,
+                    CommandOptionType.NoValue);
+
+                CommandOption fingerPrint = verifyCmd.Option(
+                    "--certificate-fingerprint",
+                    Strings.VerifyCommandCertificateFingerprintDescription,
+                    CommandOptionType.MultipleValue);
+
+                CommandOption verbosity = verifyCmd.Option(
+                    "-v|--verbosity",
+                    Strings.Verbosity_Description,
+                    CommandOptionType.SingleValue);
+
+                verifyCmd.HelpOption(XPlatUtility.HelpOption);
+                verifyCmd.Description = Strings.VerifyCommandDescription;
+
+                verifyCmd.OnExecute(async () =>
+                {
+                    ValidatePackagesPath(packagePaths);
+
+                    VerifyArgs args = new VerifyArgs();
+#if IS_DESKTOP
+                    args.PackagePath = packagePaths.Value;
+#else
+                    args.PackagePaths = packagePaths.Values;
+#endif
+                    args.Verifications = all.HasValue() ?
+                        new List<Verification>() { Verification.All } :
+                        new List<Verification>() { Verification.Signatures };
+                    args.CertificateFingerprint = fingerPrint.Values;
+                    args.Logger = getLogger();
+                    setLogLevel(XPlatUtility.MSBuildVerbosityToNuGetLogLevel(verbosity.Value()));
+
+                    var runner = getCommandRunner();
+                    var verifyTask = runner.ExecuteCommandAsync(args);
+                    await verifyTask;
+
+                    return verifyTask.Result;
+                });
+            });
+        }
+        private static void ValidatePackagesPath(CommandArgument argument)
+        {
+            if (argument.Values.Count == 0 ||
+                argument.Values.Any<string>(packagePath => string.IsNullOrEmpty(packagePath)))
+            {
+                throw new ArgumentNullException(argument.Name);
```

Comment:
argument null seems the incorrect exception here. 

Is this supposed to be user facing? 
The error message could be better.

Reply:
> Is this supposed to be user facing?
Yes. I will change the error message.

Reply:
Changed the error message `Unable to verify package. Argument '<package-paths>' not provided. `. Reused existing entry from [resource file](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.CommandLine.XPlat/Strings.resx#L376).